docs/design/async-checkpoint.md

```diff
@@ -0,0 +1,176 @@
+# Fastly Save and Load Checkpoint to Resume Training
+
+The design to introduce how to speed up saving and loading the checkpoint
+to resume training.
+
+## Background
+
+It is very difficult to keep training a fundational model during a lone time
+without any interruption. Hareware failure, OS failure or network breakdwon
+may oftern happen during the training. Users usually periodically checkpoint
+the model and optimizer states to the storage to resume the training
+after an interruption. Now, It takes a few minutes to use  `torch.save` to
+checkpoint a large fundational model. Frequent checkpointing may waste
+many hours of the accelerator (GPU, NPU, TPU).
+Because, the training must stop until the checkpointing is completed.
```

Comment:
`accelerator (GPU, NPU, TPU). Because, the training` - > `accelerator (GPU, NPU, TPU) since the training`